File: bank-service/src/main/java/rs/edu/raf/dto/CreditTypeDTO.java
package rs.edu.raf.dto;

import java.math.BigDecimal;

public record CreditTypeDTO(Long id, String name, BigDecimal nominalInterestRate, int minLoanTerm, int maxLoanTerm,
                            BigDecimal maxLoanAmount, BigDecimal prepayment) {
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/dto/CreditTypeDTO.kt
package org.example.bankservicekotlin.dto

import java.math.BigDecimal

data class CreditTypeDTO(
    val id: Long?,
    val name: String,
    val nominalInterestRate: BigDecimal,
    val minLoanTerm: Int,
    val maxLoanTerm: Int,
    val maxLoanAmount: BigDecimal,
    val prepayment: BigDecimal,
    val currencyDTO: CurrencyDTO
)
